File: create_standalone.py
import base64
import os

def image_to_base64(image_path):
    with open(image_path, 'rb') as image_file:
        encoded_string = base64.b64encode(image_file.read()).decode('utf-8')
    return f"data:image/jpeg;base64,{encoded_string}"

# Read the original HTML file
with open('index.html', 'r') as file:
    html_content = file.read()

# Get all images from the folder
image_folder = 'images anvi test2'
image_files = [
    'image1.JPG', 'image2.JPG', 'image3.JPG', 'image4.JPG', 'image5.JPG',
    'image6.JPG', 'image7.JPG', 'image8.JPG', 'image9.JPG', 'image10.jpg',
    'image11.JPG', 'image12.JPEG'
]

# Create the images array with base64 data
images_array = []
for i, image_file in enumerate(image_files, 1):
    image_path = os.path.join(image_folder, image_file)
    if os.path.exists(image_path):
        base64_data = image_to_base64(image_path)
        images_array.append(f"{{ id: {i}, url: '{base64_data}', rating: 1200 }}")

# Replace the IMAGES array in the HTML
images_array_str = "let IMAGES = [\n    " + ",\n    ".join(images_array) + "\n];"
html_content = html_content.replace("let IMAGES = [", images_array_str)

# Write the new HTML file
with open('index_standalone.html', 'w') as file:
    file.write(html_content)

print("Standalone version created successfully!") 

File: create_standalone.js
const fs = require('fs');
const path = require('path');

// Function to convert image to base64
function imageToBase64(imagePath) {
    const image = fs.readFileSync(imagePath);
    return `data:image/jpeg;base64,${image.toString('base64')}`;
}

// Read the original HTML file
const htmlContent = fs.readFileSync('index.html', 'utf8');

// Get all image paths
const imagePaths = [
    'images anvi test2/image1.JPG',
    'images anvi test2/image2.JPG',
    'images anvi test2/image3.JPG',
    'images anvi test2/image4.JPG',
    'images anvi test2/image5.JPG',
    'images anvi test2/image6.JPG',
    'images anvi test2/image7.JPG',
    'images anvi test2/image8.JPG',
    'images anvi test2/image9.JPG',
    'images anvi test2/image10.jpg',
    'images anvi test2/image11.JPG',
    'images anvi test2/image12.JPEG'
];

// Convert images to base64 and create the images array
const imagesArray = imagePaths.map((path, index) => {
    const base64 = imageToBase64(path);
    return `{ id: ${index + 1}, url: '${base64}', rating: 1200 }`;
});

// Replace the image loading code with the pre-converted images
const newHtmlContent = htmlContent.replace(
    /let IMAGES = \[\];\s*async function initializeImages\(\) {[\s\S]*?initializeImages\(\);}/,
    `let IMAGES = [\n    ${imagesArray.join(',\n    ')}\n];\n\nfunction init() {\n    console.log('Starting application...');\n    loadState();\n    generateComparisons();\n    showNextComparison();\n    setupKeyboardControls();\n}`
);

// Write the new HTML file
fs.writeFileSync('index_standalone.html', newHtmlContent);

console.log('Standalone version created successfully!'); 